### ISSUE TITLE
Fail to convert weather data to solar power for sites with latitude < 0

### ISSUE THREAD

**Issue by WesleyDAN** [PyPSA/atlite#103]
An error, saying 'Not implemented for negative latitudes', was raised when I converted weather data in regions with negative latitudes. 
Can Atlite still get the right converted results after the 'abs()' function is added to 'orientation.py'? Cheers.

**Comment by euronion** [comment]
You are referring to these lines, right?

https://github.com/PyPSA/atlite/blob/64a74439a99d9e11e27bb554f369282c2a302ad6/atlite/pv/orientation.py#L41-L54

Changed to
```
below_25 = np.abs(lat.values) <= np.deg2rad(25) 
below_50 = np.abs(lat.values) <= np.deg2rad(50) 
```

should work, the simple approach applied there does in its original form not distinguish between northern and southern hemisphere.

But that's also exactly what makes me wary and a bit reluctant to change it:
Maybe there is some good reason for this limitation (but maybe the assert is obsolete).

@coroa Did you write that part of the code and remember why the limitation was put into place?

**Comment by WesleyDAN** [comment]
Exactly, but I am not sure whether 'azimuth=180.' should be changed to 'azimuth=0.' . because it is explained that  'azimuth = 180 faces exactly South' at the beginning in 'orientation.py'.

**Comment by euronion** [comment]
Yep, that has to be addressed as well.

Can you checkout the branch `optimal-orientation-globally` and see whether this works for you and produces expected results?

**Comment by WesleyDAN** [comment]
Thanks, it's difficult to check out whether the result is correct. 
But I am not sure 'azimuth = np.empty_like(lat.values)' will create expected array according to the sign of value (it creates random value, eg create -5 for latitude 20 instead of 5)
Also, it seems 'abs' is not applied to negative latitude, in which case, the equation produce incorrect panel slopes.
 
I replace 'azimuth = np.empty_like(lat.values) with 'lat_copy = np.copy(lat.values),
azimuth  = (lat_copy>0)-0.5', 
and add 'lat = abs(lat)' before 'below_25 … '

Does it make sense please?

**Comment by euronion** [comment]
I think I get what you mean. ;-)

Doing multiple things simultaneously I forgot to incoporate the `np.abs(...)`.
The initial values in the `azimuth` variable are just placeholders and we need it as an empty shell. Masking is done with the original latitude values.

It should work now with the changes in the new branch. I tested it locally with reasonable results.

It would be great if you can test it for yourself and see if the results are reasonable!
Afterwards we can incorporate it into the productive code.

If you do not know how, then one method is to download the repository and checkout the new branch.
Using the console:
```
git clone https://github.com/PyPSA/atlite
git checkout remotes/origin/optimal-orientation-globally
```
Create a new environment for the test with the required packages, e.g.:
```
conda create --name atlite-test
conda activate atlite-test
conda env update -f atlite/atlite/environment.yaml
```
and install the branch as a package
```
pip install -e atlite/
```
Finally you need to edit the `atlite/atlite/config.py` file to point to the correct folder for your existing cutouts.
(as you originally did after the installation)

You should then be able to run your script as you are used to.

**Comment by WesleyDAN** [comment]
Many thanks for your timely reply.
1) Yes, it will work when using 'azimuth[lat.values <= 0] = 0', instead of 'azimuth[azimuth <= 0] = 0'
2) I am afraid (e.g.) 'slope[below_25] = 0.87 * lat.values[below_25]' will still return negative values/slopes. The equation for calculating optimal panel slope only works correctly when positive value is fed according to the reference.
   'lat.values = np.abs(lat.values)' before 'below_25 = np.abs(lat.values) <= np.deg2rad(25)' should be added, right?

**Comment by euronion** [comment]
Thanks attentive!
The slope calculation is now always positive (as you correctly pointed out it should be):

https://github.com/PyPSA/atlite/blob/1a9338c94b2117ed68fab3c4f343ef3977441a2c/atlite/pv/orientation.py#L49-L51

Changing `lat.values = np.abs(lat.values)` would change the original `lat` object (call-by-reference/function call with a mutuable argument). So that's not an option.

**Comment by WesleyDAN** [comment]
Many thanks. Yes, it will work correctly now.

**Comment by euronion** [comment]
Can you test the branch locally (as I described above)? It would be helpful to have a positive test before incoporating it into the main code.

**Comment by WesleyDAN** [comment]
Yes, no problem. I will check out whether it produces the same results with my current codes.

**Comment by WesleyDAN** [comment]
Hi, sorry for the delay. In order to test, I created a new enviroment and installed 'git', 'pip', then I follow the steps as you provied. 

It seems I cannot clone using  'git clone https://github.com/PyPSA/atlite', always shown 'RPC failed; curl 18 transfer closed with outstanding read data remaining' or 'fatal: destination path atlite already existes and is not an empty diretory'. I'm afraid it is the problem of my laptop's configuration or sth else, not the code. Btw, it was a tough work as well when I installed the existing current 'atlite', so it is the least option for me to re-install or update it honestly.

Fortunately, waht we wanna do mostly is to check part of the code in 'orientation.py' for converting at sites with negative latitude. I have applied the code for dozens of sites in the southern hemisphere, and the code can always produce reasonable results.
